poly-synth/src/poly-synth-demo.js

```diff
@@ -22,163 +22,368 @@ class PolySynthDemo {
+            1, t);
+        this.polySynth_.activeBitcrusherRoute.gain.linearRampToValueAtTime(
+            0, t);
+
+        bitcrusherBitDepthSlider_.disable();
+        bitcrusherReductionSlider_.disable();
+      } else {
+        event.target.textContent = 'Active';
+        const t = this.context_.currentTime + 0.01;
+        this.polySynth_.activeBitcrusherRoute.gain.linearRampToValueAtTime(
+            1, t);
+        this.polySynth_.bypassBitcrusherRoute.gain.linearRampToValueAtTime(
+            0, t);
+
+        bitcrusherBitDepthSlider_.enable();
+        bitcrusherReductionSlider_.enable();
+      }
+    }
+  }
+
+  initializeConvolverGUI_(containerId, selectorId) {
+    let convolverWetnessSlider_ = new ParamController(
+        containerId, this.polySynth_.setConvolverWetness.bind(this.polySynth_),
+        {
+          name: 'Wetness',
+          id: 'wetness',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.1,
+          default: this.polySynth_.convolverWetness
+        });
+
+    // The synth's convolver node buffer changes depending the selected url.
+    let selector = document.getElementById(selectorId);
+    for (let index in this.impulseResponseUrls_) {
+      let urlParts = this.impulseResponseUrls_[index].split('/');
+      let abbreviatedUrl = urlParts[urlParts.length - 1];
+      let option = document.createElement('option');
+      option.value = index;
+      option.textContent = abbreviatedUrl;
+      selector.appendChild(option);
+    }
+
+    selector.onchange = (event) => {
+      let responseIndex = parseInt(event.target.value);
+      this.polySynth_.setConvolverBuffer(
+          this.impulseResponseBuffers_[responseIndex]);
+
+      // Deselect the target to prevent interference with keyboard.
+      event.target.blur();
+    }
+  }
+
+  initializeFeedbackDelayGUI_(containerId){
+    let feedbackWetnessSlider_ = new ParamController(
+        containerId, this.polySynth_.setFeedbackWetness.bind(this.polySynth_), {
+          name: 'Wetness',
+          id: 'wetness',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.1,
+          default: this.polySynth_.feedbackWetness
+        });
+
+    let feedbackDelaySlider_ = new ParamController(
+        containerId, this.polySynth_.setFeedbackDelayTime.bind(this.polySynth_), {
+          name: 'Delay',
+          id: 'delay',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.1,
+          default: this.polySynth_.feedbackDelayTime
+        });
+
+    let feedbackGainSlider_  = new ParamController(
+        containerId, this.polySynth_.setFeedbackGain.bind(this.polySynth_), {
+          name: 'Gain',
+          id: 'feedbackGain',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.1,
+          default: this.polySynth_.feedbackGain
+        });
+  }
+
+  getImpulseResponseUrls_() {
+    let impulseResponseUrls =
```

Comment:
Basically every time you call this method, an array with a bunch of strings will be created. Is that what we want?